source/references/glossary.markdown

```diff
@@ -84,166 +129,228 @@ See [external node classifier](#external-node-classifier).
 
 ### idempotent
 
-Able to be applied multiple times with the same outcome. Puppet resources are idempotent, since they describe a desired final state rather than a series of steps to follow.
-
-(The only major exception is the `exec` type; exec resources must still be idempotent, but it's up to the user to design each exec resource correctly.)
+**Idempotence** refers to the concept of doing something multiple times with the same outcome. Puppet [resources](#resource) are idempotent, since they describe a desired final state rather than a series of steps to follow.
 
+The most prominent exception among Puppet resources is the `exec` [resource type](#type), which should be idempotent but rely on the user to design them accordingly.
 
 ### inheritance (class)
 
-A Puppet class can be derived from one other class with the `inherits` keyword. The derived class will declare all of the same resources, but can override some of their attributes and add new resources.
+A Puppet [class](#class) can be derived from another class by using the `inherits` keyword. The derived class [declares](#declare) all of the same [resources](#resource), but can override some of their [attributes](#attribute) and add new resources. Inheritance should be used very sparingly; for more information, see the [Classes](/puppet/latest/reference/lang_classes.html#inheritance) page in the [Puppet language](#puppet-language) reference.
+
+### lambda
 
-> **Note:** Most users should avoid inheritance most of the time. Unlike object-oriented programming languages, inheritance isn't terribly important in Puppet; it is only useful for overriding attributes, which can be done equally well by using a single class with a few [parameters](#parameter-defined-types-and-parameterized-classes).
+(or **code block**)
 
+A **lambda** is a block of parameterized [Puppet language](#puppet-language) code that you can pass to certain [functions](#function). For more information, see the [Lambdas (Code Blocks)](/puppet/latest/reference/lang_lambdas.html) page in the Puppet language reference.
 
+### local scope
+
+See [scope](#scope).
 
-### inheritance (node)
+### main manifest
 
-Node statements can be derived from other node statements with the `inherits` keyword. This works identically to the way class inheritance works.
+(or **site manifest**)
 
-> **Note:** Node inheritance **should almost always be avoided.** Many new users attempt to use node inheritance to look up variables that have a common default value and a rare specific value on certain nodes; it is not suited to this task, and often yields the opposite of the expected result. If you have a lot of conditional per-node data, we recommend using the Heira tool or assigning variables with an ENC instead.
+The **main manifest** is the main "point of entry" [manifest](#manifest) used by a Puppet [master](#master) when [compiling](#compilation) a [catalog](#catalog). A main manifest specific to a [site](#site) is called a **site manifest**.
```

Comment:
OK. We should scrub the docs to use or prioritize main manifest over site manifest.